packages/web-frontend/src/selectors/measureSelectors.js

```diff
@@ -34,139 +32,143 @@ const displayInfoCache = createCachedSelector(
     return {
       organisationUnitCode,
       ...data,
-      ...getMeasureDisplayInfo({ ...data }, options, hiddenMeasures),
+      ...getMeasureDisplayInfo(data, options, hiddenMeasures),
     };
   },
 )((_, __, ___, organisationUnitCode) => organisationUnitCode);
 
-const getOrgUnitFromMeasureData = (measureData, code) =>
+const getMeasureDataByOrgUnit = (measureData, code) =>
   measureData.find(val => val.organisationUnitCode === code);
 
 const selectDisplayInfo = (measureOptions, hiddenMeasures, measureData, organisationUnitCode) =>
   safeGet(displayInfoCache, [measureOptions, hiddenMeasures, measureData, organisationUnitCode]);
 
-export const selectCurrentMeasureIds = createSelector(
-  [selectCurrentMapOverlayIds, state => state.mapOverlayBar.mapOverlayHierarchy],
-  (currentMapOverlayIds, mapOverlayHierarchy) => {
-    const hierarchy = flattenMapOverlayHierarchy(mapOverlayHierarchy);
-
-    return hierarchy.map(mapOverlay =>
-      currentMapOverlayIds.includes(mapOverlay.mapOverlayId) ? mapOverlay.measureIds : [],
-    );
+export const selectMeasureOptions = createSelector(
+  [state => state.map.measureInfo, (_, mapOverlayIds) => mapOverlayIds],
+  (measureInfo = {}, mapOverlayIds) => {
+    const selectedMeasureOptions = mapOverlayIds.reduce((results, mapOverlayId) => {
+      const { measureOptions = [] } = measureInfo[mapOverlayId] || {};
+      return [...results, ...measureOptions];
+    }, []);
+
+    const measureOptionsByKey = selectedMeasureOptions.reduce((results, measureOption) => {
+      const { key } = measureOption;
+      return results[key] ? results : { ...results, [key]: measureOption };
+    }, {});
```

Comment:
Don't override existing measureOption if same key.
e.g.: two map overlays have same measure, we don't want to show the same legend in map and same item in flippa table.